sfs/mono/drivingfunction.py

```diff
@@ -673,6 +673,7 @@ def _sph_hn2(n, z):
     return jn - 1j * yn
 
 
-def _max_order_circular_harmonics(N):
+def _max_order_circular_harmonics(N, max_order):
     """Compute order of 2D HOA."""
-    return N // 2 if N % 2 == 0 else (N - 1) // 2
+    return (max_order if max_order is not None
+            else N // 2 if N % 2 == 0 else (N - 1) // 2)
```

Comment:
I really like if/else expressions, but as soon as they need more than one lines, they lose a bit of their appeal ...

How about this:

```python
    if max_order is not None:
        return max_order
    return ...
```

And am I crazy or is the last line exactly equivalent to simply `N // 2`?

If yes, we can go back to the nice one-liner:

```python
return max_order if max_order is not None else N // 2
```

Or even better:

```python
return N // 2 if max_order is None else max_order
```

I overlooked that when I changed it in 39787214e6808ecaa2910ec2159925ef98fbe0de.